content/en/tutorials/lotus/store-and-retrieve/set-up.md

```diff
@@ -111,15 +106,71 @@ You can install Lotus on MacOS 10.11 El Capitan or higher. You must have [Homebr
 
 1. Lotus is now installed on your computer.
 
-### Ubuntu
+[Head onto the next section to run your Lotus lite-node ↓](#run-a-lotus-lite-node)
+
+### MacOS build from source
+
+If Homebrew doesn't work for you, or if you prefer to build from source, try these steps. 
+
+1. Check that the XCode command line tools are installed.
+
+    ```shell
+    xcode-select -p (if missing, run xcode-select --install )
+    ```
+1. Install the pre-requisites via Homebrew
+    ```shell
+    brew install go bzr jq pkg-config rustup hwloc
+    ```
+1. Clone the latest sources
+    ```shell
+    git clone https://github.com/filecoin-project/lotus.git
+    ```
+1. Switch into the lotus folder
+    ```shell
+    cd lotus/
+    ```
+1. Checkout the latest release
+    ```shell
+    git checkout tags/v1.16.1
+    ```
+1. Setup some environment variables correctly
+    ```shell
+    export LIBRARY_PATH=/opt/homebrew/lib; export FFI_BUILD_FROM_SOURCE=1
+    ```
+1. Install Rust (when prompted, choose the default 'install' option) 
+    ```shell
+    rustup-init
+    ```
+
+1. Build the clients
+    ```shell
+    make all
+    ```
```

Comment:
Make sure there's an empty line on either side of a code block:

````
1. Checkout the latest release:

    ```shell
    git checkout tags/v1.16.1
    ```

1. Setup some environment variables correctly:

    ```shell
    export LIBRARY_PATH=/opt/homebrew/lib; export FFI_BUILD_FROM_SOURCE=1
    ```

````

Reply:
done